geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientHealthMonitorIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+
+  /**
+   * Creates and starts the server instance
+   */
+  private int createServer() {
+    CacheServer server = null;
+    try {
+      Properties p = new Properties();
+      // make it a loner
+      p.put(MCAST_PORT, "0");
+      p.put(LOCATORS, "");
+
+      this.system = DistributedSystem.connect(p);
+      this.cache = CacheFactory.create(system);
+      server = this.cache.addCacheServer();
+      int port = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+      server.setMaximumTimeBetweenPings(TIME_BETWEEN_PINGS);
+      server.setMaxThreads(getMaxThreads());
+      server.setPort(port);
+      server.start();
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail("Failed to create server");
+    }
+    return server.getPort();
+  }
+
+  @Test
+  public void settingMonitorIntervalViaProperty() {
+    int monitorInterval = 10;
+    System.setProperty(ClientHealthMonitor.CLIENT_HEALTH_MONITOR_INTERVAL_PROPERTY,
+        String.valueOf(monitorInterval));
+
+    assertEquals(monitorInterval,
+        ClientHealthMonitor
+            .getInstance(mock(InternalCache.class), 0, mock(CacheClientNotifierStats.class))
+            .getMonitorInterval());
+  }
+
+  @Test
+  public void monitorIntervalDefaultsWhenNotSet() {
+    assertNotNull(ClientHealthMonitor
+        .getInstance(mock(InternalCache.class), 0, mock(CacheClientNotifierStats.class))
+        .getMonitorInterval());
+  }
+
+  @Test
+  public void monitorIntervalDefaultsWhenInvalidValue() {
+    String monitorInterval = "this isn't a number";
+    System.setProperty(ClientHealthMonitor.CLIENT_HEALTH_MONITOR_INTERVAL_PROPERTY,
+        monitorInterval);
+
+    assertNotNull(ClientHealthMonitor
+        .getInstance(mock(InternalCache.class), 0, mock(CacheClientNotifierStats.class))
+        .getMonitorInterval());
+  }
+
+  /**
+   * This test performs the following:<br>
+   * 1)create server<br>
+   * 2)initialize proxy object and create region for client<br>
+   * 3)perform a PUT on client by acquiring Connection through proxy<br>
+   * 4)stop server monitor threads in client to ensure that server treats this as dead client <br>
+   * 5)wait for some time to allow server to clean up the dead client artifacts<br>
+   * 6)again perform a PUT on client through same Connection and verify after the put that the
+   * Connection object used was new one.
+   */
+  @Test
+  public void testDeadClientRemovalByServer() throws Exception {
+    System.setProperty(ClientHealthMonitor.CLIENT_HEALTH_MONITOR_INTERVAL_PROPERTY,
+        String.valueOf("100"));
+    PORT = createServer();
+    createProxyAndRegionForClient();
+    StatisticsType statisticsType = this.system.findType("CacheServerStats");
+    final Statistics statistics = this.system.findStatisticsByType(statisticsType)[0];
+    assertEquals(0, statistics.getInt("currentClients"));
+    assertEquals(0, statistics.getInt("currentClientConnections"));
+    this.system.getLogWriter()
```

Comment:
Any reason to not log to stdout?

Reply:
Not sure, this is actually just a rename of the original JUnitTest (apparently github decided to show it as new code even though git seemed to recognize the move when I committed it).  That being the case I don't want to mess with this too much.